editor/src/clj/editor/scene_tools.clj

```diff
@@ -457,22 +457,31 @@
           (manip-rotate evaluation-context node-id local-rotation))))
     (:scale-x :scale-y :scale-z :scale-xy :scale-xz :scale-yz)
     (fn [start-pos pos]
-      (let [start-delta (doto (Vector3d.) (.sub start-pos manip-pos))
+      (let [world-transform (:world-transform (peek original-values))
+            start-delta (doto (Vector3d.) (.sub start-pos manip-pos))
             delta (doto (Vector3d.) (.sub pos manip-pos))
-            div-fn (fn [v ^Double sv] (if (> (Math/abs sv) math/epsilon) (/ v sv) 1.0))
-            scale-factor (Vector3d.
-                           (case manip
-                             (:scale-x :scale-xy :scale-xz)
-                             (div-fn (.x delta) (.x start-delta))
-                             1.0)
-                           (case manip
-                             (:scale-y :scale-xy :scale-yz)
-                             (div-fn (.y delta) (.y start-delta))
-                             1.0)
-                           (case manip
-                             (:scale-z :scale-xz :scale-yz)
-                             (div-fn (.z delta) (.z start-delta))
-                             1.0))]
+            axis-scale (fn [axis]
+                         (let [local (case axis
+                                       :x (Vector3d. 1.0 0.0 0.0)
+                                       :y (Vector3d. 0.0 1.0 0.0)
+                                       :z (Vector3d. 0.0 0.0 1.0))]
+                           (.normalize local)
```

Comment:
Normalizing here doesn't do anything since `local` is already a unit vector at this point. Did you mean to normalize after the `.transform` call? That would make sense, since the `world-transform` may be non-orthographic when non-uniformly scaled parents are combined with rotations.